### PR TITLE
Update section on API docs, mock, tests

### PR DESCRIPTION
**Pull request by timgent**

### CODE REVIEW

**Review by danielepolencic — APPROVED** [review]
LGTM 👍

File: README.adoc [hunk 1]
```diff
@@ -58,9 +58,42 @@ Basic HTTP auth requires a username and password to be sent with every request.
...
+
+*http://projects.spring.io/spring-restdocs/[Spring Restdocs]* +
+Spring Restdocs allows you to generate documentation snippets for each of your tests. It records the request and the response and creates snippets for each of these. You can then embed these snippets into a master document that describes your whole API.
+
+Documentation snippets are recorded in asciidoctor format, which is like a more advanced version of markdown. The main readme asciidoc file can then embed snippets from other files
+
+*http://wiremock.org/[Wiremock]* +
+Wiremock is a mock API that allows you to record a series of requests and responses (by proxying requests through wiremock). When running your rest-assured tests you should proxy requests through a wiremock instance in record mode. It will then generate a number of files that represent the requests and responses. These files can then be used to run wiremock as a mock, where it will then respond with the recorded responses.
+
+=== Alternative approach - swagger
```
Comment (easternbloc) [inline README.adoc]: maybe add a link here?

File: README.adoc [hunk 2]
```diff
@@ -58,9 +58,42 @@ Basic HTTP auth requires a username and password to be sent with every request.
...
+Swagger is a very common way to document APIs. It does a reasonable job of getting some API documentation available. The shortcomings are that it can become out of sync (it relies heavily on annotations which can get out of date), the mock you can generate from a swagger spec is very inflexible, and the docs are also very inflexible. The major benefits are that it is a common format that you can put together very quickly.
+
+=== Alternative approach - API Blueprint
```
Comment (easternbloc) [inline README.adoc]: and here

**Review by easternbloc — CHANGES_REQUESTED** [review]
looks good apart from those links...

File: README.adoc (hunk 2 above)
Comment (timgent) [reply README.adoc]: Done

File: README.adoc (hunk 1 above)
Comment (timgent) [reply README.adoc]: Done

**Comment by timgent** [comment]
Thanks guys, @easternbloc those changes are done now

**Review by easternbloc — APPROVED** [review]

**Comment by easternbloc** [comment]
👍

**Pull request merged**